plugins/storage/object/minio/src/main/java/org/apache/cloudstack/storage/datastore/driver/MinIOObjectStoreDriverImpl.java

```diff
@@ -250,20 +258,28 @@ public void deleteBucketPolicy(String bucketName, long storeId) {
 
     }
 
+    protected void updateAccountCredentials(final long accountId, final String accessKey, final String secretKey) {
+        Map<String, String> details = new HashMap<>();
+        details.put(MINIO_ACCESS_KEY, accessKey);
+        details.put(MINIO_SECRET_KEY, secretKey);
+        _accountDetailsDao.persist(accountId, details);
+    }
+
     @Override
     public boolean createUser(long accountId, long storeId) {
         Account account = _accountDao.findById(accountId);
         MinioAdminClient minioAdminClient = getMinIOAdminClient(storeId);
-        String accessKey = "acs-"+account.getAccountName();
+        String accessKey = getUserOrAccessKeyForAccount(account);
         // Check user exists
         try {
             UserInfo userInfo = minioAdminClient.getUserInfo(accessKey);
             if(userInfo != null) {
-                s_logger.debug("User already exists in MinIO store: "+accessKey);
+                s_logger.info(String.format("User already exists in MinIO store: %s, updating credentials", accessKey));
+                updateAccountCredentials(accountId, accessKey, userInfo.secretKey());
                 return true;
             }
         } catch (Exception e) {
-            s_logger.debug("User does not exist. Creating user: "+accessKey);
+            s_logger.debug("User does not exist. Creating user: " + accessKey);
```

Comment:
```suggestion
            s_logger.debug(String.format("User does not exist. Creating user: %s", accessKey), e);
```
or error level?

Reply:
@DaanHoogland I don't think we need error level here. Based on the code I understand ideally user shouldn't exist on the object store we create one. Though I'm not sure if we should catchall

Reply:
Made some changes to error handling